assignment3/GridWordSearcher.java

```diff
@@ -0,0 +1,121 @@
+package assignment3;
+
+import java.util.*;
+
+public class GridWordSearcher {
+    private int n;

```

Comment:
Just using one letter variable names here is very unclear for readability. I can't tell what it is counting. 
But also, why extract these to a field? It is probably clearer just to use grid.length and grid[0].length everywhere you need them.